analyzer/server.go

```diff
@@ -108,18 +109,24 @@ func (s *Server) ListenAndServe() {
 		defer wg.Done()
 
 		addr, err := net.ResolveUDPAddr("udp", s.Addr+":"+strconv.FormatInt(int64(s.Port), 10))
-		conn, err := net.ListenUDP("udp", addr)
+		s.Conn, err = net.ListenUDP("udp", addr)
 		if err != nil {
 			panic(err)
 		}
-		defer conn.Close()
+		defer s.Conn.Close()
 
-		s.handleUDPFlowPacket(conn)
+		s.handleUDPFlowPacket(s.Conn)
```

Comment:
I think the connection should not be passed as argument as you change it to a class member. So no argument must be applied.